crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -874,18 +874,20 @@ where
                 },
             ) => {
                 // Visit the default arguments, but avoid the body, which will be deferred.
-                for ParameterWithDefault {
```

Comment:
All the changes in `crates/ruff` are fairly mechanical -- just reacting to the fact that this field is now optional.